python/test/t_QuadraticLeastSquares_std.py

```diff
@@ -1,54 +1,63 @@
-point[0] = 0.5
-point[1] = -0.5
-data[7] = point
-point[0] = 0.5
-point[1] = 0.5
-data[8] = point
-myLeastSquares = ot.QuadraticLeastSquares(data, myFunc)
-myLeastSquares.run()
-responseSurface = myLeastSquares.getMetaModel()
-print("myLeastSquares=", repr(myLeastSquares))
-print("responseSurface=", repr(responseSurface))
-inPoint = ot.Point(myFunc.getInputDimension(), 0.2)
-print("myFunc(", repr(inPoint), ")=", repr(myFunc(inPoint)))
-print("responseSurface(", repr(inPoint), ")=", repr(responseSurface(inPoint)))
-dataOut = myFunc(data)
-myLeastSquares = ot.QuadraticLeastSquares(data, dataOut)
-myLeastSquares.run()
-responseSurface = myLeastSquares.getMetaModel()
-print("myLeastSquares=", repr(myLeastSquares))
-print("responseSurface=", repr(responseSurface))
-inPoint = ot.Point(myFunc.getInputDimension(), 0.2)
-print("myFunc(", repr(inPoint), ")=", repr(myFunc(inPoint)))
-print("responseSurface(", repr(inPoint), ")=", repr(responseSurface(inPoint)))
+X = ot.Sample([[0.5] * 2, [-1.0] * 2, [-1.0, 1.0], [1.0, -1.0], [1.0] * 2, [-0.5] * 2, [-0.5, 0.5], [0.5, -0.5], [0.5] * 2])
+Y = f(X)
+algo = ot.QuadraticLeastSquares(X, Y)
+algo.run()
+mm = algo.getMetaModel()
+print("algo=", repr(algo))
+print("metamodel=", mm)
+x = [0.2] * 2
+y = f(x)
+y2 = mm(x)
+print(f"f(x)={y}")
+print(f"mm(x)={y2}")
+ott.assert_almost_equal(y2, [0.075251, 0.833126, 1.29782])
+ott.assert_almost_equal(algo.getConstant(), [0.00674598, 0.917414, 1.28186])
+ott.assert_almost_equal(algo.getLinear()[0, 0], 0.00202379)
+ott.assert_almost_equal(algo.getLinear()[0, 1], -0.00364348)
+ott.assert_almost_equal(algo.getQuadratic()[0, 0, 0], -0.00674598)
+ott.assert_almost_equal(algo.getQuadratic()[0, 1, 0], 1.69913)
+
+# 1-d
+X = [[1.0], [2.0], [3.0], [4.0]]
+Y = [[5.0], [3.0], [4.0], [7.0]]
+algo = ot.QuadraticLeastSquares(X, Y)
+algo.run()
+mm = algo.getMetaModel()
+x = [0.0]
+y2 = mm(x)
+print(f"mm(x)={y2}")
+print(f"constant={algo.getConstant()}")
+print(f"linear={algo.getLinear()}")
+print(f"quadratic={algo.getQuadratic()}")
+ott.assert_almost_equal(mm.gradient([0.0])[0, 0], -5.55)
+ott.assert_almost_equal(mm.hessian([0.0])[0, 0, 0], 2.5)
+
+# 2-d
+f = ot.SymbolicFunction(["x1", "x2"], ["9*x1^2 + 6*x2^2 + 0.1*x1 * x2 + 4*x1 + 7*x2 + 5.0"])
+X = ot.Normal([2.0, 3.0], [1.0, 1.0]).getSample(1000)
+Y = f(X)
+algo = ot.QuadraticLeastSquares(X, Y)
+algo.run()
+mm = algo.getMetaModel()
+x = [2.0, 3.0]
+y = f(x)
+y2 = mm(x)
+print(f"f(x)={y}")
+print(f"mm(x)={y2}")
+print(f"constant={algo.getConstant()}")
+print(f"linear={algo.getLinear()}")
+print(f"quadratic={algo.getQuadratic()}")
+ott.assert_almost_equal(y, y2, 1e-2, 1e-2)
+ott.assert_almost_equal(algo.getLinear()[0, 0], 4.0, 1e-2, 1e-2)
+ott.assert_almost_equal(algo.getLinear()[1, 0], 7.0, 1e-2, 1e-2)
+ott.assert_almost_equal(0.5 * algo.getQuadratic()[0, 0, 0], 9.0, 1e-2, 1e-2)
+ott.assert_almost_equal(0.5 * algo.getQuadratic()[0, 1, 0], 0.1, 1e-2, 1e-2)
+ott.assert_almost_equal(0.5 * algo.getQuadratic()[1, 1, 0], 6.0, 1e-2, 1e-2)
```

Comment:
Here, the `getLinear` and `getQuadratic` methods are evaluated 3 and 4 times respectively. A single call is required.